src/main/java/net/atos/zac/aanvraag/ProductaanvraagService.java

```diff
@@ -270,15 +275,20 @@ public ProductaanvraagDimpact getProductaanvraag(final ORObject productaanvraagO
         }
     }
 
-    private void addInitiator(final String bsn, final URI zaak, final URI zaaktype) {
+    private void addInitiator(final Betrokkene betrokkene, final URI zaak, final URI zaaktype) {
         final RolType initiator = ztcClientService.readRoltype(RolType.OmschrijvingGeneriekEnum.INITIATOR, zaaktype);
-        final RolNatuurlijkPersoon rolNatuurlijkPersoon = new RolNatuurlijkPersoon(
-                zaak,
-                initiator,
-                ROL_TOELICHTING,
-                new NatuurlijkPersoon(bsn)
-        );
-        zrcClientService.createRol(rolNatuurlijkPersoon);
+        final Rol<?> rol = getRol(betrokkene, zaak, initiator);
+        zrcClientService.createRol(rol);
+    }
+
+    private Rol<?> getRol(final Betrokkene betrokkene, final URI zaak, final RolType rolType) {
+        if (StringUtils.isNotBlank(betrokkene.getInpBsn()))
+            return new RolNatuurlijkPersoon(zaak, rolType, ROL_TOELICHTING, new NatuurlijkPersoon(betrokkene.getInpBsn()));
+        if (StringUtils.isNotBlank(betrokkene.getVestigingsNummer()))
+            return new RolVestiging(zaak, rolType, ROL_TOELICHTING, new Vestiging(betrokkene.getVestigingsNummer()));
+        if (StringUtils.isNotBlank(betrokkene.getInnNnpId()))
+            return new RolNietNatuurlijkPersoon(zaak, rolType, ROL_TOELICHTING, new NietNatuurlijkPersoon(betrokkene.getInnNnpId()));
+        return null;
```

Comment:
How is this null handled further on the code? Returning nulls is often not the best option. Maybe here but not sure. I think in this case it is an exceptional case right? If so we should throw an exception instead (and log that exception of course) of returning null.

Reply:
I've gone a slightly different direction, because I think we need to process the productaanvraag as much as possible. I've added two more types of Roles, and we now log the unexpected case that we get another type of Role. The signature is now more transparent.